### PR TITLE
Initial gem implementation

### PR DESCRIPTION
**Pull request by jordanmaguire**
These view helpers have existed in one form or another in the last 5 apps I've worked on (CCIWA and HForce both have them and I'm currently working on both.

See README for demo of what it's all aboot:

``` ruby
# When rendering a nil object, no matter what type you'll get
render_data(nil) # => <span class="text-muted">N/A</span>

# Date
render_data(Date.new(2016, 4, 29), :date) # => 2016-04-29
render_data(Date.new(2016, 4, 29), :date, {format: :my_format}) # => "29/4/2016"

# DateTime
render_data(DateDateTime.new(2016, 4, 29, 17, 30), :date) # => Fri, 29 April 2016 17:30:00 +0000
render_data(DateTime.new(2016, 4, 29, 17, 30), :date, {format: :my_format}) # => "29/4/2016 5:30PM"

# Percentage
render_data(55, :percentage) # => 55%
render_data(55, :percentage, precision: 0) # => 55%
render_data(55, :percentage, precision: 1) # => 55.0%

# Strings
render_data("Jordan") # => "Jordan"
```

### CODE REVIEW

File: .rspec [hunk 1]
```diff
@@ -0,0 +1 @@
+--color
```
Comment (pzi) [inline .rspec]: naughty :)

File: README.md [hunk 2]
```diff
@@ -1,2 +1,32 @@
...
+render_data(DateTime.new(2016, 4, 29, 17, 30), :date, {format: :my_format}) # => "29/4/2016 5:30PM"
+
+# Percentage
+render_data(55, :percentage) # => 55%
+render_data(55, :percentage, precision: 0) # => 55%
+render_data(55, :percentage, precision: 1) # => 55.0%
+
+# Strings
+render_data("Jordan") # => "Jordan"
+
```
Comment (pzi) [inline README.md]: WS

**Comment by pzi** [comment]
I like it 👍

File: lib/frontier_data_renderer/view_helper.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      time_tag(value.to_date, opts.reverse_merge(format: :default))
+    when :percentage
+      number_to_percentage(value, opts.reverse_merge(precision: 0))
+    else
+      value
+    end
+  end
+
+  def no_data_message
+    content_tag(:span, "N/A", class: "text-muted")
```
Comment (pzi) [inline lib/frontier_data_renderer/view_helper.rb:27]: Maybe expose a config setting to be able to override the class for `nil` data?

Comment (jordanmaguire) [inline lib/frontier_data_renderer/view_helper.rb:27]: Going to at some point, yes. Might hook that up on weekend.

File: .ruby-version [hunk 4]
```diff
@@ -0,0 +1 @@
+2.3.0
```
Comment (sheldongill) [inline .ruby-version:1]: Can we make this 2.2.4, so it's level with AWS current?

**Comment by BenR1312** [comment]
i had to ask directly what this was doing, and its a pretty cool way to handle some quick formatting and giving some useful information when the values are nil.

i like it, i just didnt get it at first!

👍

File: lib/frontier_data_renderer/view_helper.rb [hunk 5]
```diff
@@ -0,0 +1,30 @@
+module FrontierDataRenderer::ViewHelper
+
+ def render_data(value, type=:string, opts={})
+    if value.present?
+      format_value(value, type, opts)
+    else
+      no_data_message
```
Comment (BenR1312) [inline lib/frontier_data_renderer/view_helper.rb:7]: this is the part i didnt notice initially.
is good!

File: .ruby-version (hunk 4 above)
Comment (jordanmaguire) [inline .ruby-version:1]: This only effects the development environemnt - doesn't get released with the gem

File: .rspec (hunk 1 above)
Comment (jordanmaguire) [inline .rspec]: fuxed

File: README.md (hunk 2 above)
Comment (jordanmaguire) [inline README.md]: fuxed

**Pull request merged**